src/wp-includes/block-template-areas.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Block template areas functions.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Registers the navigation areas supported by the current theme. The expected
+ * shape of the argument is:
+ * array(
+ *     'primary'   => 'Primary',
+ *     'secondary' => 'Secondary',
+ *     'tertiary'  => 'Tertiary',
+ * )
+ *
+ * @param array $new_areas Supported navigation areas.
+ */
+function register_navigation_areas( $new_areas ) {
+	global $navigation_areas;
+	$navigation_areas = $new_areas;
+}
+
+// Register the default navigation areas.
+register_navigation_areas(
```

Comment:
This should be attached to some hook like `setup_theme` that is hooked in in `default-filters.php`.